edx/edx-enterprise
# ENT-284: Refine SAP SuccessFactors course export flow

**Description:**

This pull request finalizes a couple features to do with the SAP course export feature. Specifically:

1. Retrieve the course language and use it in the submission
2. Create a course track selection URL for the main link, rather than the marketing URL
3. Send up to 1000 courses at once to SAP

**JIRA:** [ENT-284](https://openedx.atlassian.net/browse/ENT-284)

**Dependencies:** None

**Merge deadline:** 10 April 2017

**Installation instructions:**
Requires:
- Course Discovery
    - At least one course
    - At least one catalog containing at least one course
- Enterprise customer
    - Attached to catalog required above
- Configured SapSuccessFactorsGlobalConfiguration (see document provided offline)
- Configured SapSuccessFactorsEnterpriseCustomerConfiguration (see document provided offline)
    - Link to our existing EnterpriseCustomer

**Testing instructions:**
1. Run the following (where `staff` is a local superuser with permissions to the course catalog):
    ```sh
    ./manage.py lms transmit_courseware_data --catalog_user staff --settings=devstack
    ```
2. In the stdout logging of your devstack, observe the following facts:
    - The `locale` key under `description` and `title` should (roughly) match the language entered for the course in the Course Discovery stack, unless the language selected there isn't available. In that case, it should be `English`. Change this selection around a bit, and rerun the command to be sure it tracks expected behavior.
    - Under `content`, the single present item should have a `launch_url` that corresponds to the course track selection page, and which matches the domain of the site linked to your EnterpriseCustomer. Scheme will be `https` by default, unless you set the `SUCCESSFACTORS_COURSE_EXPORT_DEFAULT_URL_SCHEME` environment variable before running the command. Example:
         ```
        https://example.com/course_modes/choose/course-v1:edX+fancy+demo123/
        ```

**Merge checklist:**

  **DO NOT** just add dependencies to `requirements/*.txt` files.

**Post merge:**

**Author concerns:**

1. Because I was sick of seeing three uncovered lines, I also wrote tests for them. 100% coverage!
